### ISSUE TITLE
Read mail module and Usermin all UTF-8 subjects appears as broken apart

### ISSUE THREAD

**Issue by regs01** [webmin/webmin#1216]
Happens in both Webmin and Usermin. Happens to all emails no matter if locale is set to EN UTF-8 or RU UTF-8. Filters do not work on them.

Subject in message source
`
Subject: =?UTF-8?B?0KLQtdGB0YLQvtCy0L7QtSDRgdC+0L7QsdGJ0LXQvdC40LUg0LTQu9GPINC/0YDQvtCy?=
	=?UTF-8?B?0LXRgNC60Lgg0LrQvtC00LjRgNC+0LLQutC4INC30LDQs9C+0LvQvtCy0LrQsA==?=
`

<img width="596" alt="firefox_2020-03-13_20-41-54" src="https://user-images.githubusercontent.com/10468217/76651038-d7d69f00-6574-11ea-9dfa-30d02355f284.png">
<img width="429" alt="firefox_2020-03-13_20-42-41" src="https://user-images.githubusercontent.com/10468217/76651041-d9a06280-6574-11ea-81b7-7ad1576df764.png">

**Comment by iliarostovtsev** [comment]
Hi,

Thanks for your feedback. It's known issue and will addressed in the future, when work on mail is continued.

**Comment by 4n0n1mo** [comment]
Any updates?

**Comment by iliarostovtsev** [comment]
Can you attach an email that you're having an issue with?

**Comment by 4n0n1mo** [comment]
> Can you attach an email that you're having an issue with?

My language is Portuguese - Brazil. When I am using email filtering by subject and the subject has special characters like ç, ã, á, â, etc, the filter doesn't work.

Example:
Subject: Ação necessária : seu item foi rejeitado

Then I click the button to create an email filter, select by subject and save it, but it won't work. If I remove the special charecters and use words that doesn't contains it, then it work. It's really a problem with these special chars.

![image](https://user-images.githubusercontent.com/5435164/142892239-d596ce59-6a13-47e9-976e-74579aed1af1.png)
![image](https://user-images.githubusercontent.com/5435164/142892385-909bbfb0-4ff7-447d-b3d4-55520e960b9c.png)

**Comment by iliarostovtsev** [comment]
It would be helpful if you downloaded an email as text file and attached here as a zip.

Also, what Webmin and Usermin versions are you using exactly?

**Comment by 4n0n1mo** [comment]
Webmin version | 1.981 
Usermin version | 1.823

It also happens with other emails that has special characters in the subject.
Here is the file for this email (in three formats):
[Email.zip](https://github.com/webmin/webmin/files/7582770/Email.zip)

**Comment by iliarostovtsev** [comment]
These are not plain text messages.

By the way, the latest Usermin version is 1.830!

**Comment by 4n0n1mo** [comment]
Here it is in txt and html:
[Files.zip](https://github.com/webmin/webmin/files/7583546/Files.zip)

**Comment by iliarostovtsev** [comment]
At first, try installing Usermin 1.830 and see if the problem persists.

If it is, go to Usermin, click the button to show raw message, right click and save it as a text file, and later attach it here. What you've sent before isn't a raw message format.

**Comment by 4n0n1mo** [comment]
Sorry, I didn't know how to get it raw, now I do.
My website is an ecommerce, so I'll have to wait until 3-4am to reboot it and update.
But here it is anyways.

[view_mail.txt](https://github.com/webmin/webmin/files/7585210/view_mail.txt)

**Comment by iliarostovtsev** [comment]
> My language is Portuguese - Brazil. When I am using email filtering by subject and the subject has special characters like ç, ã, á, â, etc, the filter doesn't work.

I missed the fact that filtering itself doesn't work. Well, the problem is that _procmail_ doesn't decode any encoded strings inside of the mail message, so the solution could be is to use encoded variant of the string for matching, by checking the raw message content first.

For example, based on your message above, you could use `QcOnw6NvIG5lY2Vzc8Ohcmlh` string for filtering message subject.

**Comment by 4n0n1mo** [comment]
Thanks for your help!
It would be nice if usermin could handle it automatically, but I'll use that method in the meantime

**Comment by jcameron** [comment]
Yeah, to properly support this, we'd need to do the same encoding that's used in SMTP for the `Subject` line when creating a filter in procmail. Unclear if this will work properly for substring matches though..

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
